ts/Core/Legend/LegendSymbol.ts

```diff
@@ -100,22 +100,31 @@ namespace LegendSymbol {
 
         let attr: SVGAttributes = {},
             legendSymbol,
-            markerOptions = options.marker;
+            markerOptions = options.marker,
+            symbolOffset = 0,
+            lineSizer;
 
         // Draw the line
         if (!this.chart.styledMode) {
             attr = {
-                'stroke-width': options.lineWidth || 0
+                'stroke-width': options.lineWidth || 0,
+                'stroke-linecap': (options.linecap || 'round') as any
```

Comment:
The `any` cast is needed here because [SVGAttributes['stroke-linecap']](https://github.com/highcharts/highcharts/blob/98e732a40b8737d6199615c181adc2a3b9675a91/ts/Core/Renderer/SVG/SVGAttributes.d.ts#L99) is typed as specific strings, while `series.options.linecap` [may be a string](https://github.com/highcharts/highcharts/blob/98e732a40b8737d6199615c181adc2a3b9675a91/ts/Core/Series/SeriesOptions.d.ts#L89). If we remove the string, and also make the override in the gauge series stricter, we don't need the `any` cast.